src/pages/gateway/extending-unified-schema.md

```diff
@@ -0,0 +1,351 @@
+        }
+      }
+    ],
+    "additionalResolvers": [
+      "./additional-resolvers.js"
+    ],
+    "files": [
+      {
+        "path": "./additional-resolvers.js",
+        "content": "\r\nmodule.exports = {\r\n\tresolvers: {\r\n\t\tConfigurableProduct: {\r\n\t\t\tspecial_price: {\r\n\t\t\t\tselectionSet: \"{ name price_range { maximum_price { final_price { value } } } }\",\r\n\t\t\t\tresolve: (root, args, context, info) => {\r\n\t\t\t\t\tlet max = 0;\r\n\r\n\t\t\t\t\ttry {\r\n\t\t\t\t\t\tmax = root.price_range.maximum_price.final_price.value;\r\n\t\t\t\t\t} catch (e) {\r\n\t\t\t\t\t\t// ignore\r\n\t\t\t\t\t}\r\n\r\n\t\t\t\t\treturn context.DiscountsAPI.Query.discounts(\r\n\t\t\t\t\t\t{ root, args, context, info, selectionSet: \"{ name discount }\" }\r\n\t\t\t\t\t)\r\n\t\t\t\t\t\t.then((response) => {\r\n\t\t\t\t\t\t\tconst discountConfig = response.find((discount) => discount.name === root.name);\r\n\r\n\t\t\t\t\t\t\tif (discountConfig) {\r\n\t\t\t\t\t\t\t\treturn max * ((100 - discountConfig.discount) / 100);\r\n\t\t\t\t\t\t\t} else {\r\n\t\t\t\t\t\t\t\treturn max\r\n\t\t\t\t\t\t\t}\r\n\t\t\t\t\t\t})\r\n\t\t\t\t\t\t.catch(() => {\r\n\t\t\t\t\t\t\treturn null;\r\n\t\t\t\t\t\t});\r\n\t\t\t\t},\r\n\t\t\t},\r\n\t\t},\r\n\t},\r\n};\r\n"
+      }
+    ]
+  }
+}
+```
+
+The previous example contains a `files` object that contains the following `javascript`:
+
+```js
+module.exports = {
+	resolvers: {
+		ConfigurableProduct: {
+			special_price: {
+				selectionSet: '{ name price_range { maximum_price { final_price { value } } } }',
+				resolve: (root, args, context, info) => {
+					let max = 0;
+
+					try {
+						max = root.price_range.maximum_price.final_price.value;
+					} catch (e) {
+						// ignore
+					}
+
+					return context.DiscountsAPI.Query.discounts({
+							root,
+							args,
+							context,
+							info,
+							selectionSet: '{ name discount }',
+						})
+						.then(response => {
+							let discount = 0;
+							const discountConfig = response.find(discount => discount.name === root.name);
+
+							if (discountConfig) {
+								discount = discountConfig.discount;
+							}
+
+							return max * ((100 - discount) / 100);
+						})
+						.catch(() => {
+							return null;
+						});
+				},
+			},
+		},
+	},
+};
+```
+
+This `javascript` file creates a `special_price` field that uses the `maximum_price` for a product and then applies the discount listed for that product in the [`discountsapi.json` file](https://raw.githubusercontent.com/AdobeDocs/graphql-mesh-gateway/main/src/pages/_examples/discounts-api.json). The following arguments define how the `javascript` interacts with your mesh:
+
+- `special_price` - the name of the new field we are creating
+
+- `selectionSet` - the information selected from the source that you want to mutate
+
+- `root`, `context`, and `info` - mandatory parameters that are forwarded from the resolvers calling the method
+
+- `args` - arguments to pass to the mutation or query
```

Comment:
```suggestion
```

```js
context.DiscountsAPI.Query.discounts({
    root,
    args,
    context,
    info,
    selectionSet: '{ name discount }',
})
```
calls the `discounts` query on the `DiscountsAPI` source through the `context` object. This results in a promise which resolves to the `discounts` data.